File: serve.js
// node serve.js

import fs from 'fs'
import http from 'http'

const folder = 'a'
const port = 1234
const types = {
  css: 'text/css',
  html: 'text/html',
  ico: 'image/x-icon',
  js: 'application/javascript'
}

function prepare (request, response) {
  let file = folder + request.url
  if (!(fs.existsSync(file) && fs.lstatSync(file).isFile())) file = folder + '/x.html'
  const type = types[file.split('.').pop()]
  if (type) response.writeHead(200, {'content-type': type})
  fs.createReadStream(file).pipe(response)
}

http.createServer((request, response) => prepare(request, response)).listen(port)

console.log('localhost:' + port)

File: serve.py
# python3 serve.py

import socket

folder = 'a'
port = 1234
types = {
    'css': 'text/css',
    'html': 'text/html',
    'ico': 'image/x-icon',
    'js': 'application/javascript'
}

server = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
server.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)
server.bind(('localhost', port))
server.listen()

print('localhost:' + str(port))

while True:
    client = server.accept()[0]
    try:
        file = client.recv(1024).decode()[4:].split(' H')[0]
        ftype = types.get(file.split('.')[-1])
        if not ftype:
            file = '/x.html'
            ftype = 'text/html'
        with open(folder + file, 'rb') as file:
            client.sendall(b'HTTP/1.\ncontent-type:' + ftype.encode() + b'\n\n' + file.read())
    except:
        next
    client.close()